packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -282,6 +283,76 @@ export async function* OpenAICompletionModel(
   return AI.AppendOnlyStream;
 }
 
+function estimateFunctionTokenCount(functions: Record<string, FunctionDefinition>): number {
+  // According to https://community.openai.com/t/how-to-calculate-the-tokens-when-using-function-call/266573
+  // function definitions are serialized as TypeScript. We'll use JSON-serialization as an approximation (which
+  // is almost certainly an overestimate).
```

Comment:
The OpenAI response body reports how many tokens it counted in the input/output. We could use that to get a sense of how far we're over-estimating / check our calculations generally.

Reply:
Sadly they don't for streaming responses 😦